### PR TITLE
Update fixtures and tests with relay-compiler v10.0.0

### LINKED ISSUE #195: Relay 10: Error when compiling new directives
### Description
After updating relay packages versions to v10, I am unable to use new directives `appendEdge`, `prependEdge` and `deleteRecord`.

### Usage
For the following query:
```ts
  const [commitCreateProfile] = useMutation<
    CreateCreateProfileMutation
  >(graphql`
    mutation CreateCreateProfileMutation($input: CreateProfileInput!) {
      createProfile(input: $input) {
        result @appendEdge(connections: ["Normal_profiles"]){
          node {
            id
            category
            subcategories
            status
          }
        }
        error {
          message
        }
      }
    }
  `);
```
`relay-compiler` throws an error:
```
C:\code\work\kinderway\frontend>yarn relay
yarn run v1.18.0
$ yarn run relay-compiler --schema ../schema.graphql --src ./src/ --language typescript
$ C:\code\work\kinderway\node_modules\.bin\relay-compiler --schema ../schema.graphql --src ./src/ --language typescript
Writing ts
ERROR:
Unknown directive 'appendEdge'.
error Command failed with exit code 100.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
error Command failed with exit code 100.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

### Package versions
```
"react-relay": "^10.0.0",
"relay-runtime": "^10.0.0",
"babel-plugin-relay": "^10.0.0",
"relay-compiler": "^10.0.0",
"relay-compiler-language-typescript": "^12.0.3",
"relay-config": "^10.0.0",
"relay-test-utils": "^10.0.0",
```

`yarn why output`:
```
C:\code\work\kinderway>yarn why relay-compiler
yarn why v1.18.0
[1/4] Why do we have the module "relay-compiler"...?
[2/4] Initialising dependency graph...
[3/4] Finding dependency...
[4/4] Calculating file sizes...
=> Found "relay-compiler@9.1.0"
info Has been hoisted to "relay-compiler"
info Reasons this module exists
   - "workspace-aggregator-83504b82-9adb-45fa-83e4-d3d172b23434" depends on it
   - Hoisted from "_project_#@graphql-codegen#typescript#@graphql-codegen#visitor-plugin-common#@graphql-tools#relay-operation-optimizer#relay-co
mpiler"
   - Hoisted from "_project_#backend#graphql-tools#@graphql-tools#relay-operation-optimizer#relay-compiler"
info Disk size without dependencies: "3.55MB"
info Disk size with unique dependencies: "12.95MB"
info Disk size with transitive dependencies: "3.55MB"
info Number of shared dependencies: 142
=> Found "frontend#relay-compiler@10.0.0"
info This module exists because "_project_#frontend" depends on it.
Done in 1.93s.
```

### PR DESCRIPTION
**Pull request by wadamek65**
Even though the tests with new directives pass just fine, actually using the new directives with these changes still does not work. 

Submitting draft as I'm not sure what to check next.

Should fix #195 after it is done properly.

@sibelius

### CODE REVIEW

File: src/TypeScriptGenerator.ts [hunk 1]
```diff
@@ -1084,7 +1084,7 @@ function getDataTypeName(name: string): string {
 }
 
 // Should match FLOW_TRANSFORMS array
-// https://github.com/facebook/relay/blob/v6.0.0/packages/relay-compiler/language/javascript/RelayFlowGenerator.js#L621-L627
+// https://github.com/facebook/relay/blob/v10.0.0/packages/relay-compiler/language/javascript/RelayFlowGenerator.js#L982
```
Comment (wadamek65) [inline src/TypeScriptGenerator.ts:1087]: Only updated the url (or maybe shouldn't have?). The transforms array seems to be in sync with `relay-compiler`.

**Comment by sibelius** [comment]
do you have a custom handler?

**Comment by wadamek65** [comment]
By handler do you mean `handlerProvider`? Then no.

**Comment by sibelius** [comment]
make sure the handler is being called

https://github.com/facebook/relay/blob/master/packages/relay-runtime/handlers/RelayDefaultHandlerProvider.js#L28

and that generated have the metadata

**Comment by wadamek65** [comment]
By handler being called you mean my own repo from my issue or in `relay-compiler-language-typescript` ? If you mean my repo I don't think it has anything to do with the error, as I get it during relay compilation, not runtime. I can't even generate relay artifacts. Added the default handler explicitly just to test and I still get the error:
```
import {
  RelayNetworkLayer,
  uploadMiddleware,
} from 'react-relay-network-modern';
import {DefaultHandlerProvider, Environment, RecordSource, Store } from 'relay-runtime';

const network = new RelayNetworkLayer([uploadMiddleware()]);
const store = new Store(new RecordSource());

export const environment = new Environment({ handlerProvider: DefaultHandlerProvider, network, store });
```

**Comment by wadamek65** [comment]
Okay, I've figured it out.

I had multiple packages installed that depended on `graphql-tools` which in turn installed relay-optimizer and it uses `relay-compiler` v9. `yarn` insisted on using that binary instead of the `v10.0.0` one which resulted in that error. After symlinking it to a correct binary it now works. Still no code completion for the directives but at least the compilation step works properly.

If the PR is of any use I'd be happy to finish it.

**Comment by sibelius** [comment]
it is working before this changes?

**Comment by wadamek65** [comment]
Yes it does. This package is unrelated to my problem.

**Comment by thicodes** [comment]
Have you tested the changes in the examples?

**Comment by wadamek65** [comment]
Seems like it is working just fine (apart from the code completion). Can't confirm 100% as I'm not quite sure how to go about translating a previous config like this:
```ts
          {
            type: 'RANGE_ADD',
            parentID: userData.id,
            edgeName: 'result',
            connectionInfo: [
              { key: 'Normal_profiles', rangeBehavior: 'append' },
            ],
          },
```

Into the new `appendEdge`. But it seems to be working all fine as I'm getting an error `index.js:1 Warning: [Relay][Mutation] The connection with id 'client:root:viewer:me:QWNjb3VudC41ZjBjYjdmZDRkZjViYTEyMzRhODEyMDc=:Normal_profiles' doesn't exist.` So once I actually provide it with a correct ID it should start working.

One strange thing I noticed is that `@deleteRecord` directive seems to require `ID` type and does not accept `ID!` type, even though the `RANGE_DELETE` config works just fine (unless I am misinformed). Getting error from compiler:
```
ERROR:
Invalid use of @deleteRecord on field 'id'. Expected field type ID, got ID!.

Source: views/Main/Admin/Accounts/components/DeleteDialog.tsx (8:14)
7:         result {
8:           id @deleteRecord
                ^
9:         }
```
But this is obviously unrelated to this package.

As I've said before, if this PR is of any use I'd be happy to finish it.

**Comment by sibelius** [comment]
I think it is useful to finish this

to keep everything up to date

the last part would be to run this changes on relay-examples, also testing it new directives works well there

**Comment by wadamek65** [comment]
Okay, I will work on that next at weekend :) Thank you for your help so far, I love relay and will be happy to contribute.

**Comment by wadamek65** [comment]
@sibelius @thicodes I think this is ready for review. Updated the examples and also extended them with a checkbox "Append" to showcase/test both `appendEdge` and `prependEdge`. `deleteRecord` unfortunately doesn't fit in this example so didn't add it. The new directives are working just fine.

**Comment by maraisr** [comment]
not sure if we push #199 before this or not, both seem to upgrade graphql to v15.

I think its is a more verbose PR, perhaps we can merge these two into each other? cc @alloy @sibelius

**Comment by alloy** [comment]
> not sure if we push #199 before this or not, both seem to upgrade graphql to v15.
> 
> I think its is a more verbose PR, perhaps we can merge these two into each other? cc @alloy @sibelius

Sounds good to me 👍

**Comment by thicodes** [comment]
@wadamek65 I created a PR only with the updates of the packages and fixtures. You are making great updates to the examples. We really appreciate what you've been doing. Please get the updates and I think we'll be ready to merge

**Comment by wadamek65** [comment]
I think everything should be up to date now @thicodes

**Comment by thicodes** [comment]
good job!

**Comment by alloy** [comment]
Great job @wadamek65, thanks! 🙏

**Comment by alloy** [comment]
:rocket: PR was released in v13.0.1 :rocket:

**Pull request merged**